### ISSUE TITLE
In some cases, add more version information to processing metadata

### ISSUE THREAD

**Issue by gadomski** [stac-utils/stac-task#36]
While we'd like to think that we're always running code that's been tagged w/ a version, in reality we're sometimes (often?!) running on a unreleased version. It'd be useful to provide a switch to turn on commit SHA version information in https://github.com/stac-utils/stac-task/blob/6818b9cf42029fc603ad42df9815465b0252b09d/stactask/task.py#L171. E.g. maybe:

```json
{
    "processing:software": {
        "my-awesome-task": "0.1.0 @ abcdef0123456789"
    }
}
```

**Comment by philvarner** [comment]
We need the ability to add additional metadata like this in (e.g., which version of a stactools pkg was used), but processing:software isn't the right place for it, since this is primarily for data processing rather than metadata processing. Maybe a "tooling" extension would be a good idea to cover the areas that Processing and ARD don't?

**Comment by gadomski** [comment]
Yeah, the processing extension has always felt strange for me to indicate the software that made the metadata, so I'd be interested in a tooling extension or something like it.

**Comment by philvarner** [comment]
though maybe it's a degenerate case of ARD, and we just need to provide guidance on it. Though I think the ARD folks are also abstracting their extension to account for this.
